Empire/GraphQL.php

```diff
@@ -0,0 +1,157 @@
+                ],
+                'preloadConfigRules' => [
+                    'type' => 'String',
+                    'description' => __( 'JSON of the display (blocking) rules', 'empireio' )
+                ],
+                'preloadContainersEnabled' => [
+                    'type' => 'Boolean',
+                    'description' => __(
+                        'If true, include pre-sized divs in the DOM for optimal CLS scores',
+                        'empireio'
+                    ),
+                ],
+                'preloadContainersConfig' => [
+                    'type' => 'String',
+                    'description' => __( 'JSON of the placement rules', 'empireio' )
+                ],
+                'siteId' => [
+                    'type' => 'String',
+                    'description' => __( 'Site ID for this site within Empire', 'empireio' ),
+                ],
+            ],
+        ] );
+
+        add_action( 'graphql_register_types', [$this, 'global_ads_config'] );
+    }
+
+    public function global_ads_config() {
+        \register_graphql_field( 'RootQuery', 'empireConfig', [
+            'type' => 'EmpireConfig',
+            'description' => __( 'Sitewide Configuration for Empire Platform', 'empireio' ),
+            'resolve' => function() {
+                $testEnabled = $this->empire->getEmpirePixelTestPercent() < 100 &&
+                    $this->empire->getEmpirePixelTestPercent() > 0;
+                return [
+                    'adsEnabled' => $this->empire->isEnabled(),
+                    'adsTestEnabled' => $testEnabled,
+                    'adsTestPercentEnabled' => $testEnabled ?
+                        $this->empire->getEmpirePixelTestPercent() : null,
+                    'adsTestSplitTestKey' => $testEnabled ?
+                        $this->empire->getEmpirePixelTestValue() : null,
+                    'adsTxt' => $this->empire->getAdsTxtManager()->get(),
+                    'ampAdsEnabled' => $this->empire->useAmpAds(),
+                    'connatixPlayspaceEnabled' => $this->empire->useConnatix(),
+                    'connatixPlayspaceId' => $this->empire->useConnatix() ?
+                        $this->empire->getConnatixPlayspaceId() : null,
+                    'oneTrustEnabled' => $this->empire->useCmpOneTrust(),
+                    'oneTrustSiteId' => $this->empire->getOneTrustId(),
+                    'preloadConfigEnabled' => $this->empire->useInjectedAdsConfig(),
+                    'preloadConfigRules' => $this->empire->getAdsConfig()->adRules ?
+                        json_encode($this->empire->getAdsConfig()->adRules) : '[]',
+                    'preloadContainersEnabled' => $this->empire->useAdsSlotsPrefill(),
+                    'preloadContainersConfig' => $this->empire->getAdsConfig()->forPlacement ?
+                        json_encode($this->empire->getAdsConfig()->forPlacement) : '[]',
```

Comment:
Probably it will be settled with time how and what options are exposed, but it looks like several configs are mixed up here.

Currently, we have three separate configs:
1) AdsConfig - contains raw AdConfig that is inlined on the page, placements, and adRules that are used by the plugin internally
2) AmpConfig - contains HTML blobls of AMP components for placements
3) PrefillConfig - contains HTML blobs and styles for ad-conatiners

And I think they should be exposed independently.

Reply:
That makes a lot of sense! i'll adjust